builtin/mainmenu/dlg_create_world.lua

```diff
@@ -150,11 +375,53 @@ local function create_world_buttonhandler(this, fields)
 		return true
 	end
 
+	for k,v in pairs(fields) do
+		local split = string.split(k, "_", nil, 3)
+		if split and split[1] == "flag" then
+			local setting
+			if split[2] == "mg" then
+				setting = "mg_flags"
+			else
+				setting = split[2].."_spflags"
+			end
+			-- We replaced the underscore of flag names with a dash.
+			local flag = string.gsub(split[3], "-", "_")
```

Comment:
If splits are already limited to 3 splits, in which case is this workaround needed?